src/test/run-pass/variance-iterators-in-libcore.rs

```diff
@@ -0,0 +1,17 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
```

Comment:
NIT: Name for this test is weirdly general for what it tests. Linked list iterator covariance is tested in libcollections itself for example. Perhaps we could test for zip covariance in libcoretest too?